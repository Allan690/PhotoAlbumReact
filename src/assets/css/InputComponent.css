.search {
    display: flex;
    flex-flow: row wrap;
    width: 50%;
    margin: 20px auto;
}

.input-component {
    width: 300px;
    height: 30px;
    text-indent: 10px;
}

.input-component::placeholder {
 text-indent: 10px;
}

.submit {
    width: 100px;
    margin-left: 10px;
    height: 35px;
}

.submit:hover {
    cursor: pointer;
}