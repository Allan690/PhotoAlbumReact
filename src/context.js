import React from "react";
import axios from "./axiosConfig";

const initialState = { photos: [] };
const store = React.createContext(null);
const { Provider } = store;

const reducer = (state, action) => {
  switch (action.type) {
    case "SET_PHOTOS":
      return {
        ...state,
        photos: action.payload,
      };
    default:
      return new Error();
  }
};

const AppProvider = ({ children }) => {
  const [state, dispatch] = React.useReducer(reducer, initialState);
  const fetchPhotos = async (id = 2) => {
    const result = await axios.request({
      method: "get",
      url: `/${id}/photos?_limit=20`,
    });
    const { data } = result;
    // @ts-ignore
    dispatch({
      type: "SET_PHOTOS",
      payload: data,
    });
  };

  const actions = {
    fetchPhotos,
  };

  return (
    <Provider
      value={{
        state,
        actions,
      }}
    >
      {children}
    </Provider>
  );
};

export function usePhotoState() {
  return React.useContext(store).state;
}

export function usePhotoActions() {
  return React.useContext(store).actions;
}

export { store, AppProvider };
