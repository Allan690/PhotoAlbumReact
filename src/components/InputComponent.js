import React from "react";
import "../assets/css/InputComponent.css";
import { usePhotoActions } from "../context";

const SearchComponent = () => {
  const { fetchPhotos } = usePhotoActions();
  const [inputValue, setInputValue] = React.useState("");

  React.useEffect(() => {
      fetchPhotos();
      // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <div className="search">
      <input
        className="input-component"
        type="text"
        value={inputValue}
        onChange={e => setInputValue(e.target.value)}
        placeholder="Enter the album ID"
      />
      <button className="submit" onClick={() => fetchPhotos(parseInt(inputValue))}>
        Load Photos
      </button>
    </div>
  );
};

export default SearchComponent;
