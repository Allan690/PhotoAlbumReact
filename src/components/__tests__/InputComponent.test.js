import React from 'react';
import InputComponent from '../InputComponent';
import { shallow } from 'enzyme';
import { usePhotoActions } from '../../context';
jest.mock('../../context');

describe('InputComponent', () => {
    let fetchPhotos = jest.fn();
    beforeEach(() => {
        usePhotoActions.mockReturnValue({
            fetchPhotos
        });
        jest.spyOn(React, 'useEffect');
    });

    afterEach(() => jest.resetAllMocks());

    it('renders input component', () => {
        const wrapper = shallow(<InputComponent />);
        expect(wrapper).toMatchSnapshot()
    });

    it('renders search, input and button components', () => {
        const wrapper = shallow(<InputComponent />);
        expect(wrapper.find('.search')).toHaveLength(1);
        expect(wrapper.find('input')).toHaveLength(1);
        expect(wrapper.find('button')).toHaveLength(1);
    });

    it('fetches photos on component mount', () => {
        shallow(<InputComponent />);
        expect(React.useEffect.mock.calls[0][0]).toEqual(expect.any(Function));
        expect(React.useEffect.mock.calls[0][1]).toEqual([]);
    });

    it('fetches Photos when button is clicked', () => {
        const wrapper = shallow(<InputComponent />);
        wrapper.find('input').simulate('change', { target: { value: '3'}});
        wrapper.find('button').simulate('click');
        expect(fetchPhotos).toHaveBeenCalledWith(3);
    });
});
