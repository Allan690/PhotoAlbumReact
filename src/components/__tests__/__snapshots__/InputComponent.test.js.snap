// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent renders input component 1`] = `
<div
  className="search"
>
  <input
    className="input-component"
    onChange={[Function]}
    placeholder="Enter the album ID"
    type="text"
    value=""
  />
  <button
    className="submit"
    onClick={[Function]}
  >
    Load Photos
  </button>
</div>
`;
